plugin/fanout/README.md

```diff
@@ -0,0 +1,108 @@
+  provided with the meaning as described below
+  * `tls` - no client authentication is used, and the system CAs are used to verify the server certificate
+  * `tls` **CA** - no client authentication is used, and the file CA is used to verify the server certificate
+  * `tls` **CERT** **KEY** - client authentication is used with the specified cert/key pair.
+    The server certificate is verified with the system CAs
+  * `tls` **CERT** **KEY**  **CA** - client authentication is used with the specified cert/key pair.
+    The server certificate is verified using the specified CA file
+* `tls_servername` **NAME** allows you to set a server name in the TLS configuration; for instance 9.9.9.9
+  needs this to be set to `dns.quad9.net`. Multiple upstreams are still allowed in this scenario,
+  but they have to use the same `tls_servername`. E.g. mixing 9.9.9.9 (QuadDNS) with 1.1.1.1
+  (Cloudflare) will not work.
+
+* `fail-count` is the number of subsequent failed health checks that are needed before client stops trying to connect.
+* `worker-count` is the number of parallel queries per request. By default equals to count of IP list. Use this only for reducing parallel queries per request.
+* `policy` is specifies the policy to use for selecting response from proxies.
+  * `any` is a policy that returns any first response from success connected proxies.
+  * `first-positive` is a default policy that returns any first non-negative response from proxies.
+        * `network` is a specific network protocol. Could be `tcp`, `udp`, `tcp-tls`.
+* `except` is a list is a space-separated list of domains to exclude from proxying.
+
+## Metrics
+
+If monitoring is enabled (via the *prometheus* plugin) then the following metric are exported:
+
+* `coredns_fanout_request_duration_seconds{to}` - duration per upstream interaction.
+* `coredns_fanout_request_count_total{to}` - query count per upstream.
+* `coredns_fanout_response_rcode_count_total{to, rcode}` - count of RCODEs per upstream.
+* `coredns_fanout_healthcheck_failure_count_total{to}` - number of failed health checks per upstream.
+* `coredns_fanout_healthcheck_broken_count_total{}` - counter of when all upstreams are unhealthy,
+  and we are randomly (this always uses the `random` policy) spraying to an upstream.
+
+Where `to` is one of the upstream servers (**TO** from the config), `rcode` is the returned RCODE
+from the upstream.
+
+## Examples
+Proxy all requests within `example.org.` to a nameservers running on a different ports.  The first positive response from a proxy will be provided as the result.
+
+~~~ corefile
+example.org {
+    fanout . 127.0.0.1:9005 127.0.0.1:9006 127.0.0.1:9007 127.0.0.1:9008
+}
+~~~
+
+Sends parallel requests between three resolvers, one of which has a IPv6 address via TCP. The first response from proxy will be provided as the result.
+
+~~~ corefile
+. {
+    fanout . 10.0.0.10:53 10.0.0.11:1053 [2003::1]:53 {
+        protocol TCP
```

Comment:
I consider this a debug aid, not something people need to run normally as it puts strain on the DNS system, so this should not support a lot of feature. Protocol selection is a maybe in my book, `policy` selection should be removed entirely. 
As said earlier: I'm not looking to have this a cut and paste job from *forward*. I'm fine if you think differently, but then we should make this an external plugin.

Reply:
Can you please share a usecase for debugging?

Reply:
>  selection should be removed entirely.

Thanks, done

Reply:
@miekg  I also think this plugin can be used not only for debugging.
https://github.com/coredns/coredns/issues/3653#issuecomment-583280957
I think it also can be useful for people or for some specific goals where need functionality to fanout request.

Reply:
@miekg two open questions are here:
1. What a usecase for debugging? (It could help to make changes in the plugin and add cover the usecase  by unit tests)
2. Why the functionality of this plugin can not be used by people? // related to https://github.com/coredns/coredns/issues/3653#issuecomment-583280957

Reply:
@miekg Actually, in [Network Service Mesh](https://networkservicemesh.io/) we use this not as a debugging aid, but a first class tool.  Network Service Mesh allows a workload to connect (at the level of the workload) to multiple mutually ignorant 'Network Services' simultaneously.  For example, imagine a large auto maker providing a 'partner-network-service' to thousands of its suppliers, and a parts supplier producing a workload that participates in both 'partner-network-service' from the automaker and 'manufacturing-network-service' from its manufacturing group.  Both of those have independent, mutually ignorant DNS systems.  The fanout plugin allows us to participate simultaneously in both.  Obviously in this case you *don't* want the enormous tax of having to manually coordinate those DNS systems into a single DNS system, and you also don't want the latency of having to resolve in serial.  That's what the fanout plugin solves for.  So not debugging, but rather simultaneous participation in multiple mutually ignorant DNS systems.